Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Payment Flow/PaymentFormViewController.m

```diff
@@ -82,29 +82,67 @@ - (void)savePaymentTxToParse {
+                [APIManager createTransfersWithAmount:weakSelf.totalAmount toConnectedStripeAccs:connectedStripeAccounts withSourceTxId:chargeId withBlock:^(NSError * err, NSString * transferId) {
+                    if (err) {
+                        UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not create transfer." andMessage:err.localizedDescription okCompletion:nil cancelCompletion:nil];
+                        dispatch_async(dispatch_get_main_queue(), ^{
+                            [weakSelf presentViewController:alert animated:YES completion:nil];
+                        });
+                        [queue cancelAllOperations];
+                    }
+                }];
 
-        } else {
-            NSMutableArray<NSString*> *connectedStripeAccs = [NSMutableArray array];
-            for (Nonprofit* n in self.basket.nonprofits) {
-                [connectedStripeAccs addObject:n.stripeAccountId];
             }
-            [APIManager createTransfersWithAmount:self.totalAmount toConnectedStripeAccs:connectedStripeAccs withSourceTxId:chargeId withBlock:^(NSError * err, NSString * transferId) {
-                if (err) {
-                    UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not create transfer." andMessage:err.localizedDescription okCompletion:nil cancelCompletion:nil];
-                    dispatch_async(dispatch_get_main_queue(), ^{
-                        [self presentViewController:alert animated:YES completion:nil];
-                    });
-                } else {
-                    // TODO: Update nonprofits and basket total donation value
+        }];
+    }];
+
+
+    [queue addOperationWithBlock:^{
+        [weakSelf updateTotalDonationValues];
+    }];
+
+
+}
+
+- (void)updateTotalDonationValues {
+    NSOperationQueue *queue = [NSOperationQueue new];
+    NSInteger numberOfNonprofits = self.basket.nonprofits.count;
+    NSInteger amountPerNonprofit = [self.totalAmount intValue] / numberOfNonprofits;
+    __weak typeof(self) weakSelf = self;
+    [queue addOperationWithBlock:^{
+        weakSelf.basket = [weakSelf.basket fetch];
+        weakSelf.basket.totalDonatedValue =  [NSNumber numberWithLong:[weakSelf.basket.totalDonatedValue intValue] + [weakSelf.totalAmount intValue]];
+        [self.basket saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+            if (!succeeded) {
+                [queue cancelAllOperations];
+            }
+        }];
+    }];
+    [queue addOperationWithBlock:^{
+        for (Nonprofit *n in weakSelf.basket.nonprofits) {
+            Nonprofit *nonprofit = [n fetch];
+            nonprofit.totalDonationsValue =  [NSNumber numberWithLong:[nonprofit.totalDonationsValue intValue] + amountPerNonprofit];
+            [nonprofit saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+                if (!succeeded) {
+                    [queue cancelAllOperations];
```

Comment:
Note that all these save operations are async because `saveInBackgroundWithBlock` is async, so the operation queue will finish before all the actual saves are done, and this `[queue cancelAllOperations]` call here could do nothing. You could use the synchronous variant of the save method to avoid this if it can have any unwanted side effects.